stdlib/arg.ml

```diff
@@ -319,6 +324,10 @@ let max_arg_len cur (kwd, spec, doc) =
   | _ -> max cur (String.length kwd + second_word doc)
 
 
+let replace_leading_tab s n =
+  let seen = ref false in
+  String.init n (fun i -> match s.[i] with '\t' when not !seen -> seen := true; ' ' | c -> c)
```

Comment:
I personally find the length-passing API confusing. Given the use of `(^)` below, we are apparently not looking for efficiency here, so I would just `String.map` the input string and keep the extra `String.sub` call in the `else` branch below.